file/nifile.yml

```diff
@@ -0,0 +1,816 @@
+                  in the existing set of file properties.
+                type: boolean
+              properties:
+                description: The properties to set
+                type: object
+                additionalProperties:
+                  type: string
+                example:
+                  key: value
+            required:
+              - replaceExisting
+              - properties
+      responses:
+        204:
+          description: No content
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/service-groups/Default/delete-files:
+    post:
+      tags: [files]
+      summary: Delete multiple files
+      description: >-
+        Deletes multiple files in a single API call. The request body contains
+        an array of file ids to delete.
+      operationId: DeleteMultiple
+      x-ni-operation: deleteFiles
+      x-ni-privilege: ModifyFiles
+      parameters: 
+        - in: body
+          name: files
+          description: The description of files to delete
+          required: true
+          schema:
+            description: The description of files to delete
+            title: DeleteMultipleRequest
+            type: object
+            properties:
+              ids:
+                description: The list of file IDs to delete
+                type: array
+                items:
+                  description: The ID of a file to delete
+                  type: string
+                  example: '5afb2ce3741fe11d88838cc9'
+            required: [ids]
+      responses:
+        204:
+          description: No content
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v1/service-groups/Default/query-files:
+    post:
+      tags: [files]
+      summary: Query files
+      description: >-
+        Queries the Skyline File service for a list of files that match
+        specified metadata properties.
+
+        Query elements:
+
+        - idQuery: a JSON query object with a string value.
+
+        - sizeMaxQuery: a JSON query object with an integer value.
+
+        - sizeMinQuery: a JSON query object with an integer value.
+
+        - createdQuery: a JSON query object with an ISO8601 date string value.
+  
+        - propertiesQuery: a JSON array of query objects with string values.
+      operationId: QueryAvailableFiles
+      x-ni-operation: listFiles
+      x-ni-privilege: ViewMetadata
+      x-ni-request-all-privileges: true
+      parameters:
+        - in: query
+          name: skip
+          description: How many files to skip in the result when paging.  New in version 2 of this operation.
+          type: integer
+          x-ni-data-type: U32
+          default: 0
+        - in: query
+          name: take
+          description: >-
+            How many files to return in the result, or 0 to use a default.  New in version 2 of this operation.
```

Comment:
> or 0 to use a default.

Does that mean if you put 0, it'll default to 1000?

Reply:
Yes.  A value of 0 will return 1000 files (the service-defined default).  We've changed this in other services to use -1 instead of 0 to indicate the default since taking 0 results is a valid use-case in some services.